### PR TITLE
Fix Glob.search returning paths with resolved symlinks

### PR DESCRIPTION
**Pull request by fortmarek**
Hey @davbeck 👋 

We started to use `swift-glob` at [tuist](https://github.com/tuist/tuist) and when migrating from our [previous implementation](https://github.com/tuist/tuist/blob/main/Sources/TuistSupport/Utils/Glob.swift), I noticed that `Glob.search` returns resolved symlinks.

For example, we often run our tests in a temporary directory at `/var/...`. However, `/var` is actually a symbolic link to `/private/var`. If I run `Glob.search(directory: "/var/...", include: [try Pattern("some-glob/*.swift")])`, I'd expect to get  files that begin with `/var`. But that's not the case.

The reason the symlinks get resolved is because of `FileManager.default.contentsOfDirectory` that actually resolves those symlinks during the process. The solution is to base the path on the initial URL and then append a relative path for the contents to preserve the symlink.

I couldn't find any tests for `Glob.search` but I tested the behavior at https://github.com/tuist/FileSystem. Let me know if you want me to add tests here.

### CODE REVIEW

File: Sources/Glob/GlobSearch.swift [hunk 1]
```diff
@@ -79,6 +79,9 @@ public func search(
 						var options: FileManager.DirectoryEnumerationOptions = []
 						if skipHiddenFiles {
 							options.insert(.skipsHiddenFiles)
+                            #if !os(Linux)
+                            options.insert(.producesRelativePathURLs)
+                            #endif
```
Comment (fortmarek) [inline Sources/Glob/GlobSearch.swift]: The strategy I chose doesn't work on Linux as these options are not available in the LInux version of the Foundation (thanks, Apple). Let me know what you think about the PR and how important it is to make Linux work for that scenario as well and I can try spend a bit more time into figuring this out.

**Comment by fortmarek** [comment]
I'm gonna close this as we started to use our own fork at Tuist of this repo. Let me know if you'd be interested in this.

**Comment by davbeck** [comment]
So sorry I missed this.

I would indeed be interested in merging this upstream.

In regards to Linux compatibility, I think at the very least we should have some way to highlight it. Perhaps make it an option that is marked unavailable on Linux?

**Comment by fortmarek** [comment]
> In regards to Linux compatibility, I think at the very least we should have some way to highlight it. Perhaps make it an option that is marked unavailable on Linux?

Is there a reason why we should be marking the library as unavailable on Linux instead of making it compatible? The fork we currently [have](https://github.com/tuist/FileSystem/blob/main/Sources/Glob/GlobSearch.swift#L31) does have compatibility with Linux.

No worries for missing this PR! I'll see if I'll have time to upstream the changes, but we did a bunch of changes in our fork, so might take me a while if I were honest.

**Comment by davbeck** [comment]
I just meant if Linux compatibility isn't an option. I'll take a look at your fork.

**Pull request closed without merging**